From f12659ebfa34b805d7a99778c9d86cc60377a312 Mon Sep 17 00:00:00 2001
From: Sugar Zhang <sugar.zhang@rock-chips.com>
Date: Mon, 18 Jan 2021 18:36:06 +0800
Subject: [PATCH] clk: rockchip: rv1126: Better jitter performance for audio
 rate

PLL 491520000 for audio samplerate:
    8000
   16000
   24000
   32000
   48000
   64000
   96000
  128000
  192000

  ppm ranges [-1000 : +976] for clk compensation

PLL 496742400 for audio samplerate:
   11025
   22050
   44100
   88200
  176400

  ppm ranges [-1000 : +1000] for clk compensation

e.g. UAC with 48000 SR + UVC:

&cru {
	assigned-clocks =
		<&pmucru CLK_RTC32K>, <&pmucru PLL_GPLL>,
		<&pmucru PCLK_PDPMU>, <&cru PLL_CPLL>,
		<&cru PLL_HPLL>, <&cru ARMCLK>,
		<&cru ACLK_PDBUS>, <&cru HCLK_PDBUS>,
		<&cru PCLK_PDBUS>, <&cru ACLK_PDPHP>,
		<&cru HCLK_PDPHP>, <&cru HCLK_PDAUDIO>,
		<&cru HCLK_PDCORE_NIU>;
	assigned-clock-rates =
		<32768>, <1188000000>,
		<100000000>, <491520000>,
		<1400000000>, <600000000>,
		<500000000>, <200000000>,
		<100000000>, <300000000>,
		<200000000>, <150000000>,
		<200000000>;
};

Change-Id: I96dcaf9d40fdea58a4c3a823d91720203e0a75ad
Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
---
 drivers/clk/rockchip/clk-rv1126.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/clk/rockchip/clk-rv1126.c b/drivers/clk/rockchip/clk-rv1126.c
index 9ba0bff0711a..1b95aede9f74 100644
--- a/drivers/clk/rockchip/clk-rv1126.c
+++ b/drivers/clk/rockchip/clk-rv1126.c
@@ -74,6 +74,8 @@ static struct rockchip_pll_rate_table rv1126_pll_rates[] = {
 	RK3036_PLL_RATE(594000000, 1, 99, 4, 1, 1, 0),
 	RK3036_PLL_RATE(504000000, 1, 84, 4, 1, 1, 0),
 	RK3036_PLL_RATE(500000000, 1, 125, 6, 1, 1, 0),
+	RK3036_PLL_RATE(496742400, 1, 124, 6, 1, 0, 3113851),
+	RK3036_PLL_RATE(491520000, 1, 40, 2, 1, 0, 16106127),
 	RK3036_PLL_RATE(408000000, 1, 68, 2, 2, 1, 0),
 	RK3036_PLL_RATE(312000000, 1, 78, 6, 1, 1, 0),
 	RK3036_PLL_RATE(216000000, 1, 72, 4, 2, 1, 0),
-- 
2.35.3

